pkg/anonymization/anonymizer.go

```diff
@@ -0,0 +1,308 @@
+		networkRegex := regexp.MustCompile(Ipv4NetworkRegex)
+		for _, network := range networkRegex.FindAllString(installConfig, -1) {
+			networks = append(networks, network)
+		}
+	}
+
+	// we're sorting by subnet lengths, if they are the same, we use subnet itself
+	utils.SortAndRemoveDuplicates(&networks, func(i, j int) bool {
+		if !strings.Contains(networks[i], "/") || !strings.Contains(networks[j], "/") {
+			return networks[i] > networks[j]
+		}
+
+		network1 := strings.Split(networks[i], "/")
+		network2 := strings.Split(networks[j], "/")
+
+		// subnet length
+		if network1[1] > network2[1] {
+			return true
+		} else if network1[1] < network2[1] {
+			return false
+		} else {
+			return network1[0] > network2[0]
+		}
+	})
+
+	return NewAnonymizer(baseDomain, networks)
+}
+
+// AnonymizeMemoryRecord takes record.MemoryRecord, removes the sensitive data from it and returns the same object
+func (anonymizer *Anonymizer) AnonymizeMemoryRecord(memoryRecord *record.MemoryRecord) *record.MemoryRecord {
+	if len(anonymizer.clusterBaseDomain) != 0 {
+		memoryRecord.Data = bytes.ReplaceAll(
+			memoryRecord.Data,
+			[]byte(anonymizer.clusterBaseDomain),
+			[]byte(ClusterBaseDomainPlaceholder),
+		)
+		memoryRecord.Name = strings.ReplaceAll(
+			memoryRecord.Name,
+			anonymizer.clusterBaseDomain,
+			ClusterBaseDomainPlaceholder,
+		)
+	}
+
+	memoryRecord.Data = anonymizer.ipNetworkRegex.ReplaceAllFunc(memoryRecord.Data, func(originalIPBytes []byte) []byte {
+		return []byte(anonymizer.ObfuscateIP(string(originalIPBytes)))
+	})
+
+	return memoryRecord
+}
+
+// ObfuscateIP takes an IP as a string and returns obfuscated version. If it exists in the translation table,
+// we just take it from there, if it doesn't, we create an obfuscated version of this IP
+// and record it to the translation table
+func (anonymizer *Anonymizer) ObfuscateIP(ipStr string) string {
+	if strings.Contains(ipStr, "/") {
+		// we do not touch subnets themselves
+		return ipStr
+	}
+
+	if obfuscatedIP, exists := anonymizer.translationTable[ipStr]; exists {
+		return obfuscatedIP
+	}
+
+	originalIP := net.ParseIP(ipStr)
+	if originalIP == nil {
+		klog.Warningf("Unable to parse IP '%v'", ipStr)
+		// Unable to parse an IP, so just return whatever it is. It shouldn't happen.
+		return ipStr
+	}
+
+	isIPv4 := originalIP.To4() != nil
+
+	if !isIPv4 {
+		// TODO: to be implemented later
+		// the problem is that some strings can be incorrectly identified as ip v6
+		// we can try looking only for those which are wrapped by quotes or something
+		return ipStr
+	}
+
+	// We could use something like https://github.com/yl2chen/cidranger but we shouldn't typically have many networks
+	// so it's fine to just iterate over them
+	for i := range anonymizer.networks {
+		networkInfo := &anonymizer.networks[i]
+		network := &networkInfo.network
+
+		if network.IP.Equal(originalIP) {
+			return originalIP.String()
+		}
+
+		if network.Contains(originalIP) {
```

Comment:
If I understand it correctly then we silently skip the IP if this is not true. Is that OK? I think I am missing the context.

Reply:
Nope, it means we haven't found a network for this IP and we will return 0.0.0.0 after for loop

Reply:
Got it. First I didn't realize it can be found in other network. Thanks